pyomo/core/base/constraint.py

```diff
@@ -84,14 +85,14 @@ def C_rule(model, i, j):
 
     model.c = Constraint(rule=simple_constraint_rule(...))
     """
-    return rule_wrapper(
-        rule,
-        {
-            None: Constraint.Skip,
-            True: Constraint.Feasible,
-            False: Constraint.Infeasible,
-        },
-    )
+    result_map = {None: Constraint.Skip}
+    for l_type in native_logical_types:
+        result_map[l_type(True)] = Constraint.Feasible
+        result_map[l_type(False)] = Constraint.Infeasible
+    # Note: some logical types has the same as bool (e.g., np.bool_), so
```

Comment:
I think you meant this?
```suggestion
    # Note: some logical types act the same as bool (e.g., np.bool_), so
```

Reply:
Nope - I actually meant `hash`